pkg/webhook/server/server.go

```diff
@@ -21,65 +20,75 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
 
 	"github.com/qinqon/kube-admission-webhook/pkg/certificate"
-	"github.com/qinqon/kube-admission-webhook/pkg/certificate/triple"
+	"github.com/qinqon/kube-admission-webhook/pkg/certificate/chain"
 )
 
 type Server struct {
-	webhookServer *webhook.Server
-	certManager   *certificate.Manager
-	log           logr.Logger
+	webhookServer  *webhook.Server
+	webhookConfigs map[certificate.WebhookReference]bool
+	certManager    func() (*certificate.Manager, error)
+	log            logr.Logger
 }
 
-type ServerModifier func(w *webhook.Server)
+type ServerModifier func(s *Server)
 
 // Add creates a new Conditions Mutating Webhook and adds it to the Manager. The Manager will set fields on the Webhook
 // and Start it when the Manager is Started.
-func New(client client.Client, certificateOpts certificate.Options, serverOpts ...ServerModifier) (*Server, error) {
-	certManager, err := certificate.NewManager(client, certificateOpts)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed constructing certificate manager")
-	}
-
+func New(name, namespace string, client client.Client, options chain.Options, serverOpts ...ServerModifier) (*Server, error) {
 	s := &Server{
 		webhookServer: &webhook.Server{
 			Port:    8443,
 			CertDir: "/etc/webhook/certs/",
 		},
-		certManager: certManager,
-		log:         logf.Log.WithName("webhook/server"),
+		webhookConfigs: map[certificate.WebhookReference]bool{},
+		log:            logf.Log.WithName("webhook/server"),
 	}
 	s.UpdateOpts(serverOpts...)
 	s.webhookServer.Register("/readyz", healthz.CheckHandler{Checker: healthz.Ping})
+	s.certManager = func() (*certificate.Manager, error) {
+		webhooks := []certificate.WebhookReference{}
+		for webhook := range s.webhookConfigs {
+			webhooks = append(webhooks, webhook)
+		}
+		certManager, err := certificate.NewManager(name, namespace, client, options, webhooks)
+		return certManager, err
+	}
+
 	return s, nil
 }
 
-func WithHook(path string, hook *webhook.Admission) ServerModifier {
-	return func(s *webhook.Server) {
-		s.Register(path, hook)
+func WithHook(path string, hook *webhook.Admission, config certificate.WebhookReference) ServerModifier {
```

Comment:
There is no real relation between hooks and config, in case we have the same hook and is used a two different config dos it means that we have to call `WithHook` twice, so let's add a `WithConfig`